sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -57,9 +57,20 @@ process.on("exit", (code: number) => {
     }
 });
 
+const providerCache: { [key: string]: dynamic.ResourceProvider} = {};
+
 function getProvider(props: any): dynamic.ResourceProvider {
+    const providerString = props[providerKey];
+    let provider: any;
+    if(providerCache[providerString]) {
+        provider = providerCache[providerString];
+    } else {
+        provider = requireFromString(props[providerKey]).handler();
+        providerCache[providerString] = provider;
+    }
```

Comment:
`providerString` could be quite large. I'm not sure how much V8 will optimize how it's currently written, but looking it up in the cache once instead of twice may have a measurable improvement for large provider strings, and it's less code:

```suggestion
    const providerString = props[providerKey];
    let provider: any = providerCache[providerString];
    if (!provider) {
        provider = requireFromString(providerString).handler();
        providerCache[providerString] = provider;
    }
```